src/components/TagPicker.tsx

```diff
@@ -0,0 +1,133 @@
+        ];
+    }, [selectedTag]);
+
+    const enabledTags = useMemo(() => {
+        if (!shouldShowDisabledAndSelectedOption) {
+            return policyTagList;
+        }
+        const selectedNames = selectedOptions.map((s) => s.name);
+        const tags = [...selectedOptions, ...Object.values(policyTagList).filter((policyTag) => policyTag.enabled && !selectedNames.includes(policyTag.name))];
+
+        return tags;
+    }, [selectedOptions, policyTagList, shouldShowDisabledAndSelectedOption]);
+
+    //
+    const sections = useMemo(() => {
+        const policyRecentlyUsedTagsList = policyRecentlyUsedTags?.tag ?? [];
+
+        // TODO: Remove this and one line under once OptionsListUtils (https://github.com/Expensify/App/issues/24921) is migrated to TypeScript.
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return (
+            // @ts-expect-error TODO: Remove this once OptionsListUtils (https://github.com/Expensify/App/issues/24921) is migrated to TypeScript.
+            OptionsListUtils.getFilteredOptions({}, {}, [], searchValue, selectedOptions, [], false, false, false, {}, [], true, enabledTags, policyRecentlyUsedTagsList, false).tagOptions
+        );
+    }, [searchValue, enabledTags, selectedOptions, policyRecentlyUsedTags?.tag]);
+
+    const headerMessage = OptionsListUtils.getHeaderMessageForNonUserList((sections?.[0]?.data?.length ?? 0) > 0, searchValue);
+
+    // @ts-expect-error TODO: Remove this once OptionsListUtils (https://github.com/Expensify/App/issues/24921) is migrated to TypeScript.
+    const selectedOptionKey = sections[0]?.data?.filter((policyTag) => policyTag.searchText === selectedTag)?.[0]?.keyForList;
+
+    return (
+        <OptionsSelector
+            // @ts-expect-error TODO: Remove this once OptionsSelector (https://github.com/Expensify/App/issues/25125) is migrated to TypeScript.
+            contentContainerStyles={[{paddingBottom: StyleUtils.getSafeAreaMargins(insets).marginBottom}]}
+            optionHoveredStyle={styles.hoveredComponentBG}
+            sectionHeaderStyle={styles.mt5}
+            sections={sections}
+            selectedOptions={selectedOptions}
+            headerMessage={headerMessage}
+            textInputLabel={translate('common.search')}
+            boldStyle
+            highlightSelectedOptions
+            isRowMultilineSupported
+            shouldShowTextInput={shouldShowTextInput}
+            // Focus the first option when searching
+            focusedIndex={0}
+            // Focus the selected option on first load
+            initiallyFocusedOptionKey={selectedOptionKey}
+            onChangeText={setSearchValue}
+            onSelectRow={onSubmit}
+        />
+    );
+}
+
+TagPicker.displayName = 'TagPicker';
+
+export default withOnyx<TagPickerProps, TagPickerOnyxProps>({
+    policyTags: {
+        key: ({policyID}) => `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}` as typeof ONYXKEYS.COLLECTION.POLICY_TAGS,
```

Comment:
```suggestion
        key: ({policyID}) => `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
```

Reply:
You can move with this suggestion @fvlvte , it turns out that the problem we discussed before was because you was typing `policyTags` as `OnyxCollection`, but it should be `OnyxEntry` like @VickyStash suggested.